src/sentry/seer/anomaly_detection/store_data.py

```diff
@@ -57,6 +61,60 @@ def _get_start_and_end_indices(data: list[TimeSeriesPoint]) -> tuple[int, int]:
     return start, end
 
 
+def handle_send_historical_data_to_seer(alert_rule: AlertRule, project: Project, method: str):
+    try:
+        rule_status = send_historical_data_to_seer(
+            alert_rule=alert_rule,
+            project=project,
+        )
+        if rule_status == AlertRuleStatus.NOT_ENOUGH_DATA:
+            # if we don't have at least seven days worth of data, then the dynamic alert won't fire
+            alert_rule.update(status=AlertRuleStatus.NOT_ENOUGH_DATA.value)
+    except (TimeoutError, MaxRetryError):
+        raise TimeoutError(f"Failed to send data to Seer - cannot {method} alert rule.")
+    except ParseError:
+        raise ParseError("Failed to parse Seer store data response")
+    except (ValidationError, Exception):
+        raise ValidationError(f"Failed to send data to Seer - cannot {method} alert rule.")
+
+
+def send_new_rule_data(organization: Organization, alert_rule: AlertRule, project: Project) -> None:
+    if not features.has("organizations:anomaly-detection-alerts", organization):
+        alert_rule.delete()
+        raise ResourceDoesNotExist("Your organization does not have access to this feature.")
+    try:
+        handle_send_historical_data_to_seer(alert_rule, project, "create")
+    except (TimeoutError, MaxRetryError, ParseError, ValidationError):
+        alert_rule.delete()
+        raise
+    else:
+        metrics.incr("anomaly_detection_alert.created")
+
+
+def update_rule_data(
+    organization: Organization,
+    alert_rule: AlertRule,
+    project: Project,
+    updated_fields: dict[str, Any],
+    updated_query_fields: dict[str, Any],
+) -> None:
+    if not features.has("organizations:anomaly-detection-alerts", organization):
+        raise ResourceDoesNotExist("Your organization does not have access to this feature.")
```

Comment:
nit: i tend to use these checks to ensure that the method shouldn't be invoked, then leave the internal of the method "pure" -- by doing that it's easier to do logic / flow control with many experiments running.

Reply:
Do you mean just move the check to outside of the function?